CHANGELOG.md

```diff
@@ -4,6 +4,7 @@
 
 * Fixed a crash when calling Table.toString() in debugger (#2429).
 * Fixed a crash that RealmResults was not updated in Realm's change listener by adjusting the calling orders of listeners on Realm, RealmObject and RealmResults (#2926).
+* Fixed a potential crash when accessing other RealmResults from inside a RealmResults' RealmChangeListener (#2951).
```

Comment:
Maybe merge these two items. And after this commit, the order of listeners doesn't matter too much anymore.